x-pack/plugins/stack_connectors/server/connector_types/bedrock/bedrock.ts

```diff
@@ -258,53 +281,78 @@ The Kibana Connector in use may need to be reconfigured with an updated Amazon B
   public async invokeAI({
     messages,
     model,
+    stopSequences,
+    temperature,
   }: InvokeAIActionParams): Promise<InvokeAIActionResponse> {
     const res = await this.runApi({
-      body: JSON.stringify(formatBedrockBody({ messages, model })),
+      body: JSON.stringify(formatBedrockBody({ messages, stopSequences, temperature })),
       model,
     });
     return { message: res.completion.trim() };
   }
 }
 
 const formatBedrockBody = ({
-  model = DEFAULT_BEDROCK_MODEL,
   messages,
-  stopSequences = ['\n\nHuman:'],
-  temperature = 0.5,
+  stopSequences,
+  temperature = 0,
 }: {
-  model?: string;
   messages: Array<{ role: string; content: string }>;
   stopSequences?: string[];
   temperature?: number;
-}) => {
-  const combinedMessages = messages.reduce((acc: string, message) => {
-    const { role, content } = message;
-    const [, , modelName, majorVersion, minorVersion] =
-      (model || '').match(/(\w+)\.(.*)-v(\d+)(?::(\d+))?/) || [];
-    // Claude only has Assistant and Human, so 'user' will be converted to Human
-    let bedrockRole: string;
+}) => ({
+  anthropic_version: 'bedrock-2023-05-31',
+  messages: ensureMessageFormat(messages),
+  max_tokens: DEFAULT_TOKEN_LIMIT,
+  stop_sequences: stopSequences,
+  temperature,
+});
 
-    if (
-      role === 'system' &&
-      modelName === 'claude' &&
-      Number(majorVersion) >= 2 &&
-      Number(minorVersion) >= 1
-    ) {
-      bedrockRole = '';
-    } else {
-      bedrockRole = role === 'assistant' ? '\n\nAssistant:' : '\n\nHuman:';
+/**
+ * Ensures that the messages are in the correct format for the Bedrock API
+ * Bedrock only accepts assistant and user roles.
+ * If 2 user or 2 assistant messages are sent in a row, Bedrock throws an error
+ * We combine the messages into a single message to avoid this error
+ * @param messages
+ */
+const ensureMessageFormat = (
```

Comment:
@dgieselaar I got o11y working by adding this safety function. If you are able to optimize the request on your end instead, that would be even safer. Thanks!

Reply:
yeah I'm happy to merge on our side